### PR TITLE
Culling algorithm improvements

### PR DESCRIPTION
**Pull request by Kroppeb**
Probably fixes the following 2 issues (which I think are duplicates?): #722 and #872

### CODE REVIEW

File: src/main/java/net/caffeinemc/sodium/render/chunk/occlusion/ChunkOcclusion.java [hunk 1]
```diff
@@ -1,136 +1,266 @@
...
+        }
 
-    private static int markDirection(int state, int dir) {
-        return (state | (1 << dir));
-    }
+        last++;
 
-    private static boolean hasDirection(int state, int dir) {
-        return (state & (1 << dir)) != 0;
+        if (first != 0 && last != sections.length) {
```
Comment (Leo40Git) [inline src/main/java/net/caffeinemc/sodium/render/chunk/occlusion/ChunkOcclusion.java:258]:
```suggestion
        if (first != 0 && last != sections.length - 1) {
```
At present, this "trim array" path is always run.

Comment (Kroppeb) [reply src/main/java/net/caffeinemc/sodium/render/chunk/occlusion/ChunkOcclusion.java:258]: yes and no, it's a off by one where I initialize.

Comment (Kroppeb) [reply src/main/java/net/caffeinemc/sodium/render/chunk/occlusion/ChunkOcclusion.java:258]: actually no, the `last++` fixes that.

Comment (Leo40Git) [reply src/main/java/net/caffeinemc/sodium/render/chunk/occlusion/ChunkOcclusion.java:258]: Oh, oops, sorry. This `if` just looks fishy.

**Pull request merged**